### PR TITLE
feat: [CDS-43137]: edit secrets

### PR DESCRIPTION
**Pull request by radujomir**
### Summary

Fixed issue when editing references secrets and was displayed by default edit text secret modal, 
After this fix depending on secret reference type the correct edit modal will be displayed (SSH, WinRm, Text, File)

#### Screenshots

https://user-images.githubusercontent.com/104008782/206472379-442eb75f-b0ca-4633-8a46-33d3b9c2da47.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 7 Issues 
* Code issue 7 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11247)

**Comment by radujomir** [comment]
retrigger build

**Comment by radujomir** [comment]
retrigger cypress

**Comment by radujomir** [comment]
retrigger titlecheck

**Comment by radujomir** [comment]
retrigger titlecheck

**Comment by radujomir** [comment]
retrigger titlecheck

**Comment by radujomir** [comment]
retrigger titlecheck

File: src/modules/30-secrets/components/SecretReference/SecretReference.tsx [hunk 1]
```diff
@@ -28,6 +29,9 @@ import useCreateUpdateSecretModal from '@secrets/modals/CreateSecretModal/useCre
 import { ResourceType } from '@rbac/interfaces/ResourceType'
 import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
 import RbacButton from '@rbac/components/Button/Button'
+import useCreateSSHCredModal from '@secrets/modals/CreateSSHCredModal/useCreateSSHCredModal'
```
Comment (YounixIO) [inline src/modules/30-secrets/components/SecretReference/SecretReference.tsx:32]: Update the import to use via relative path.

**Review by avinashmadhwani02 — DISMISSED** [review]

**Comment by radujomir** [comment]
retrigger cypress-rest

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/30-secrets/modals/CreateWinRmCredModal/views/StepDetails.tsx [hunk 2]
```diff
@@ -23,9 +24,23 @@ const StepWinRmDetails: React.FC<StepProps<WinRmCredSharedObj> & WinRmCredShared
...
   const { getString } = useStrings()
+  if (loading) {
+    return (
+      <Container
+        style={{
+          paddingTop: '200px',
+          width: '100%',
+          height: '500px'
+        }}
+      >
```
Comment (litvinovstas) [inline src/modules/30-secrets/modals/CreateWinRmCredModal/views/StepDetails.tsx:39]: move it in css

File: src/modules/30-secrets/modals/CreateSSHCredModal/views/StepDetails.tsx [hunk 3]
```diff
@@ -23,9 +24,23 @@ const StepSSHDetails: React.FC<StepProps<SSHCredSharedObj> & SSHCredSharedObj> =
...
   detailsData,
   authData,
-  isEdit
+  isEdit,
+  loading
 }) => {
   const { getString } = useStrings()
+  if (loading) {
+    return (
+      <Container
```
Comment (litvinovstas) [inline src/modules/30-secrets/modals/CreateSSHCredModal/views/StepDetails.tsx:33]: move styles in css

**Review by litvinovstas — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Pull request merged**